questionnaire/metadata/metadata-service.test.js

```diff
@@ -0,0 +1,173 @@
+                        type: 'metadata',
+                        id: validQuestionnaireId2,
+                        attributes: {
+                            'questionnaire-id': validQuestionnaireId2,
+                            created: '2023-06-08 00:00:00.000000+00',
+                            modified: '2023-06-08 00:00:00.000000+00',
+                            expires: '2023-06-08 00:30:00.000000+00',
+                            'submission-status': 'NOT_STARTED'
+                        }
+                    }
+                ];
+
+                const actual = await metadataService.getMetadata();
+
+                expect(Array.isArray(actual.data)).toBe(true);
+                expect(actual.data).toMatchObject(expected);
+            });
+
+            it('Should error if the apiVersion is not "2023-05-17"', async () => {
+                const metadataService = createMetadataService({
+                    logger: () => 'Logged from createQuestionnaire test',
+                    apiVersion: undefined,
+                    ownerId
+                });
+
+                await expect(metadataService.getMetadata()).rejects.toThrow(
+                    `getMetadata cannot be used with ownerId "${ownerId}" and apiVersion "undefined"`
+                );
+            });
+
+            it('Should error if the ownerId is not defined', async () => {
+                const metadataService = createMetadataService({
+                    logger: () => 'Logged from createQuestionnaire test',
+                    apiVersion,
+                    ownerId: undefined
+                });
+
+                await expect(metadataService.getMetadata()).rejects.toThrow(
+                    `getMetadata cannot be used with ownerId "undefined" and apiVersion "${apiVersion}"`
+                );
+            });
+        });
+        describe('getMetadataByQuestionnaire', () => {
+            it('Should get metadata filtered by questionnaireId and owner', async () => {
+                const expected = [
+                    {
+                        type: 'metadata',
+                        id: validQuestionnaireId,
+                        attributes: {
+                            'questionnaire-id': validQuestionnaireId,
+                            created: '2023-06-08 00:00:00.000000+00',
+                            modified: '2023-06-08 00:00:00.000000+00',
+                            expires: '2023-06-08 00:30:00.000000+00',
+                            'submission-status': 'NOT_STARTED'
+                        }
+                    }
+                ];
+
+                const actual = await metadataService.getMetadataByQuestionnaire(
+                    validQuestionnaireId
+                );
+
+                expect(Array.isArray(actual.data)).toBe(true);
+                expect(actual.data).toMatchObject(expected);
+            });
+
+            it('Should error if the apiVersion is not "2023-05-17"', async () => {
```

Comment:
Should we have a test for:
'Should error if the apiVersion is not "2023-05-17"'
'Should error if the apiVersion is undefined'